core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/dr/restore/DatabaseRestoreActions.java

```diff
@@ -0,0 +1,120 @@
+                    DatabaseRestoreTriggerEvent payload) {
+                return BackupRestoreContext.from(flowParameters, payload, payload.getBackupLocation(), payload.getBackupId());
+            }
+
+            @Override
+            protected void doExecute(BackupRestoreContext context, DatabaseRestoreTriggerEvent payload, Map<Object, Object> variables) {
+                backupRestoreStatusService.restoreDatabase(context.getStackId(), context.getBackupId());
+                sendEvent(context);
+            }
+
+            @Override
+            protected Selectable createRequest(BackupRestoreContext context) {
+                return new DatabaseRestoreRequest(context.getStackId(), context.getBackupLocation(), context.getBackupId());
+            }
+
+            @Override
+            protected Object getFailurePayload(DatabaseRestoreTriggerEvent payload, Optional<BackupRestoreContext> flowContext, Exception ex) {
+                return DatabaseRestoreFailedEvent.from(payload, ex, DetailedStackStatus.DATABASE_RESTORE_FAILED);
+            }
+        };
+    }
+
+    @Bean(name = "DATABASE_RESTORE_FINISHED_STATE")
+    public Action<?, ?> databaseRestoreFinished() {
+        return new AbstractDatabaseRestoreAction<>(DatabaseRestoreSuccess.class) {
+
+            @Override
+            protected BackupRestoreContext createFlowContext(FlowParameters flowParameters, StateContext<FlowState, FlowEvent> stateContext,
+                    DatabaseRestoreSuccess payload) {
+                return BackupRestoreContext.from(flowParameters, payload, null, null);
+            }
+
+            @Override
+            protected void doExecute(BackupRestoreContext context, DatabaseRestoreSuccess payload, Map<Object, Object> variables) {
+                backupRestoreStatusService.restoreDatabaseFinished(context.getStackId());
+                sendEvent(context);
+            }
+
+            @Override
+            protected Selectable createRequest(BackupRestoreContext context) {
+                return new StackEvent(DatabaseRestoreEvent.DATABASE_RESTORE_FINALIZED_EVENT.event(), context.getStackId());
+            }
+
+            @Override
+            protected Object getFailurePayload(DatabaseRestoreSuccess payload, Optional<BackupRestoreContext> flowContext, Exception ex) {
+                return null;
+            }
+        };
+    }
+
+    @Bean(name = "DATABASE_RESTORE_FAILED_STATE")
+    public Action<?, ?> databaseRestoreFailedAction() {
+        return new AbstractDatabaseRestoreAction<>(DatabaseRestoreFailedEvent.class) {
+
+            @Override
+            protected BackupRestoreContext createFlowContext(FlowParameters flowParameters, StateContext<FlowState, FlowEvent> stateContext,
+                    DatabaseRestoreFailedEvent payload) {
+                Flow flow = getFlow(flowParameters.getFlowId());
+                Stack stack = stackService.getById(payload.getResourceId());
+                MDCBuilder.buildMdcContext(stack);
```

Comment:
I also had to check this. So what I found:
- if you use `ErrorHandlerAwareReactorEventFactory` it will set MdcContext in the headers and flow engine will persist this
- if the above is set then `LogContextAspects` will load it and set it when a flow step is starting
- if we would like to set it, we usually do it in the context creation part, which is in the flow's abstract action in most cases, as it is the same throughout that flow

if I were you I would remove that MDCBuilder line and check the logs, and check them with it, to see if there is any difference